foundry/foundry.py

```diff
@@ -252,36 +262,68 @@ def describe(self):
         print("DC:{}".format(self.dc))
         print("Dataset:{}".format(self.dataset.json(exclude={"dataframe"})))
 
-    def publish(self, foundry_metadata, update=False, **kwargs):
+    def publish(self, foundry_metadata, data_source, title, authors, update=False, publication_year=None, **kwargs):
```

Comment:
I made it so the metadata, data_source, title, and authors were all required arguments. Does that sound right to ya'll?